### PR TITLE
bulkio: Fix `SHOW JOBS FOR SCHEDULE`

### PR DESCRIPTION
**Pull request by miretskiy**
Fix show jobs delegation code to correclty handle
`FOR SCHEDUL(s)` clause.

Release Notes: None

### CODE REVIEW

File: pkg/sql/delegate/show_jobs.go [hunk 1]
```diff
@@ -45,9 +53,6 @@ func (d *delegator) delegateShowJobs(n *tree.ShowJobs) (tree.Statement, error) {
 		// The "ORDER BY" clause below exploits the fact that all
 		// running jobs have finished = NULL.
 		orderbyClause = `ORDER BY COALESCE(finished, now()) DESC, started DESC`
-	} else if n.Schedules == nil {
-		// Limit the jobs displayed to the select statement in n.Jobs.
-		whereClause = fmt.Sprintf(`WHERE job_id in (%s)`, n.Jobs.String())
```
Comment (dt) [inline pkg/sql/delegate/show_jobs.go:50]: Don't we want this instead of the `else` below it?

**Review by miretskiy — COMMENTED** [review]
doh

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52909#-:-MExzA6VDFJ8XQ0h2Bkt:b2ni1aw)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Review by dt — APPROVED** [review]

**Pull request merged**